blacklists.py

```diff
@@ -140,17 +147,174 @@ def exists(self, item: Union[str, dict]):
                 splat = x.split('\t')
                 if len(splat) == 3 and splat[2].strip() == item:
                     return True, i
+        return False, -1
+
+
+class YAMLParserCIDR(BlacklistParser):
+    """
+    YAML parser for IP blacklist (name suggests we should move to proper CIDR eventually).
+
+    Base class for parsers for YAML files with simple schema validation.
+    """
+    # Remember to update the schema version if any of this needs to be changed
+    SCHEMA_VERSION = '2019120601'  # yyyy mm dd id
+    SCHEMA_VARIANT = 'yaml_cidr'
+    SCHEMA_PRIKEY = 'ip'
+
+    def __init__(self, filename):
+        super().__init__(filename)
+
+    def _parse(self, keep_disabled=False):
+        with open(self._filename, 'r', encoding='utf-8') as f:
+            y = yaml.safe_load(f)
+        assert y['Schema'] == self.__class__.SCHEMA_VARIANT
+        assert y['Schema_version'] <= self.__class__.SCHEMA_VERSION
+        for item in y['items']:
+            if not keep_disabled and 'disable' in item and item['disable']:
```

Comment:
Maybe `item.get('disable')` which defaults to `None` if key `disable` isn't present?

Reply:
Again, thanks for this tip.